llms/mlx_lm/server.py

```diff
@@ -496,16 +553,20 @@ def handle_stream(
         stop_sequence_suffix = None
         logging.debug(f"Starting stream:")
 
-        for (token, _), _ in zip(
+        prompt = self.get_prompt_cache(prompt)
+        prompt = mx.array(prompt)
```

Comment:
I think this is a small efficiency bug for which you added the `mx.array` on line 562.

Basically, line 600 below casts this back to a list immediately so no need to cast to array here. We can cast only when passed to `generate_step` as you do below.